lymph/edge.py

```diff
@@ -28,7 +28,8 @@ def __init__(self, start: Node, end: Node, t: float = 0.):
         self.end = end
         self.end.inc.append(self)
         self.t = t
-
+        self.states = states

```

Comment:
The `Edge` class should not have a state.

Reply:
Right now I am trying to figure out how to solve this problem in the best way. Edge assigns the transition probability based on the given transition parameters. My idea was to directly change the transition probability based on the state parent node, i.e. multiplying with the microscopic parameter if the parent node is microscopic. But this is only the case if we are in the trinary model. 
This would not be possible if edge does not know whether we are dealing with a binary or trinary model. Hence the microscopic multiplication would need to be done outside of edge, which nullifies the whole idea of putting the edge information in the edge class.
An alternative would be to change the binary model to states (0,2) then the edge class would not need to know what model we are using, but only what the parent state is.

Reply:
Maybe it's even easier: If the `Node` class has a attribute `_allowed_states = [...]`, then every `Edge` instance can very quickly and easily check which model's probability to compute. For example:

```python
def spread_prob(self) -> float:
    """Compute the probability of spread from this edge's start to its end."""
    if len(self.start._allowed_states) == 3:
        # do the trinary stuff
    else:
        # do the binary stuff
```

Of course, this could be more sophisticated. The `Node` could simply have a property

```python
@property
def is_trinary(self) -> bool:
    """Indicate if this node can take on three states."""
    return len(self._allowed_states) == 3
```

and then the `Edge` class uses this. Doing something like this is usually more elegant and safer than storing the same attribute in multiple classes, because there might be bugs that occur due to a mismatch of them. And checking for all cases that they always agree with each other is going to be a mess.